### PR TITLE
Unify parathreads and parachains in HostConfig

### PR DESCRIPTION
**Pull request by f1h125**

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2870940

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2872303

File: runtime/parachains/src/scheduler/tests.rs [hunk 1]
```diff
@@ -965,27 +964,17 @@ fn availability_predicate_works() {
...
-	} = default_config();
-	let collator = CollatorId::from(Sr25519Keyring::Alice.public());
+	let HostConfiguration { group_rotation_frequency, paras_availability_period, .. } =
+		default_config();
 
-	assert!(
-		chain_availability_period < thread_availability_period &&
-			thread_availability_period < group_rotation_frequency
-	);
+	assert!(paras_availability_period < group_rotation_frequency);
```
Comment (eskimor) [inline runtime/parachains/src/scheduler/tests.rs:970]: Not sure what this assert was about, but it is very likely no longer relevant.

File: runtime/parachains/src/configuration.rs [hunk 2]
```diff
@@ -356,12 +336,8 @@ where
...
 		}
 
-		if self.chain_availability_period.is_zero() {
-			return Err(ZeroChainAvailabilityPeriod)
-		}
-
-		if self.thread_availability_period.is_zero() {
-			return Err(ZeroThreadAvailabilityPeriod)
+		if self.paras_availability_period.is_zero() {
+			return Err(ZeroParasAvailabilityPeriod)
```
[the remark this answers is not in the document]
Comment (eskimor) [reply runtime/parachains/src/configuration.rs:340]: Probably not, was just wondering.

File: runtime/parachains/src/scheduler/tests.rs [hunk 3]
```diff
@@ -1022,39 +1007,24 @@ fn availability_predicate_works() {
...
-				.expect("predicate exists recently after rotation");
-
-			let would_be_timed_out = System::block_number() - thread_availability_period;
-
-			assert!(!pred(CoreIndex(0), would_be_timed_out)); // chains can't be timed out now.
-			assert!(pred(CoreIndex(1), would_be_timed_out)); // but threads can.
+			assert!(!pred(CoreIndex(0), now - paras_availability_period + 1));
 		}
 
-		run_to_block(1 + group_rotation_frequency + thread_availability_period, |_| None);
```
[the remark this answers is not in the document]
Comment (alexgparity) [reply runtime/parachains/src/scheduler/tests.rs:1057]: This is targeting `master` which still uses the group rotation information for the timeout calculation.

File: runtime/parachains/src/scheduler/tests.rs (hunk 1 above)
Comment (alexgparity) [reply runtime/parachains/src/scheduler/tests.rs:970]: I guess that assert was about making sure a specific ordering that the test assumes was satisfied.

**Comment by eskimor** [comment]
Hi @tdimitrov ! Could you please push this over the finish line when you are back. v6 host configuration has been released already, so we need to bump this to v7. Thank you!

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3059524

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3060719

**Comment by tdimitrov** [comment]
Currently check runtime migration jobs will fail because [this](https://github.com/paritytech/polkadot/blob/d52a8028924ef51f11543703ab80cf8af4151002/runtime/parachains/src/configuration/migration_ump.rs#L55) accesses `HostConfig` after V6 migration but before V7 one.

I think there is no point in fixing this as the UMP migration is already released. The issue is only in the `try-runtime` pre_upgrade check. Regular runtime upgrade passes (verified locally).

**Comment by ggwpez** [comment]
Hm that is a bit tricky now. The `UpdateUmpLimits` migration should always run last, since as you saw it does not have a customisable `HostConfig` type... It is also not versioned, so we need to remove it right after it executed.  
Not sure what to do now. With such a high cadence of storage migrations for the `HostConfig` it is difficult to make it work.  
We probably have to remove that migration once it executed, then it should be fine. Kusama is already in the process for .43

**Comment by tdimitrov** [comment]
We haven't got a lot of moves.
In the future we could put (artificial) `HostConfiguration` version requirement in the `UpdateUmpLimits` (or other similar migrations) and use your storage version pattern so that it passes nicely. We have to rely on the fact that it will be executed in the right time and right sequence.

But in the current case the migration is already released so we have to act swiftly, push it on Polkadot and remove it as soon as possible.

**Review by ggwpez — CHANGES_REQUESTED** [review]
The UMP limits migration must always run last, otherwise it will call into an un-migrated HostConfig.  
So going to block this MR until we removed the old migration from the code (Polkadot should soon be on .43).

**Comment by eskimor** [comment]
I believe there are other changes to `HostConfig` [here](https://github.com/paritytech/polkadot/pull/7110/files) as well, might make sense to rebase this PR to that one and merge it there.

**Comment by tdimitrov** [comment]
> I believe there are other changes to `HostConfig` [here](https://github.com/paritytech/polkadot/pull/7110/files) as well, might make sense to rebase this PR to that one and merge it there.

Done -> https://github.com/paritytech/polkadot/pull/7452

I've decided to create a new branch for the rebase to avoid the complete mess. We can close this one if we merge https://github.com/paritytech/polkadot/pull/7452

**Pull request closed without merging**